@navikt/core/react/src/form/search-field/SearchField.tsx

```diff
@@ -38,50 +24,101 @@ export interface SearchFieldProps
    * SearchField label
    */
   label: React.ReactNode;
+  /**
+   * Inverts color theme
+   * @default false
+   */
+  inverted?: boolean;
+  /**
+   * Customize aria-label on clear button
+   * @default "Slett tekst i felt"
+   */
+  clearButtonLabel?: string;
+  /**
+   * Callback for when user manually clears input with button or Escape
+   */
+  onClear?: () => void;
+  /**
+   * Callback for value in input after change
+   */
+  onChange?: (value: string) => void;
+  /**
+   * Toggles display of "clear"-button when there is text in field
+   */
+  clearButton?: boolean;
 }
 
-interface SearchFieldComponent
-  extends React.ForwardRefExoticComponent<
-    SearchFieldProps & React.RefAttributes<HTMLDivElement>
-  > {
-  Button: SearchFieldButtonType;
-  Clear: SearchFieldClearButtonType;
-  Input: SearchFieldInputType;
-}
-
-const SearchField = forwardRef<HTMLDivElement, SearchFieldProps>(
+const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
   (props, ref) => {
-    const {
-      inputProps,
-      errorId,
-      showErrorMsg,
-      hasError,
-      size,
-      inputDescriptionId,
-    } = useFormField(props, "searchfield");
+    const { inputProps, size, inputDescriptionId } = useSearchField(
+      props,
+      "searchfield"
+    );
 
     const {
       className,
       hideLabel,
-      children,
       label,
       description,
-      error,
+      value,
+      clearButtonLabel,
+      onClear,
+      inverted = false,
+      clearButton = true,
       ...rest
     } = props;
 
+    const [controlledValue, setControlledValue] = useState(value ?? "");
+    const searchRef = useRef<HTMLInputElement | null>(null);
+    const mergedRef = mergeRefs([searchRef, ref]);
+    const [wrapperRef, setWrapperRef] = useState<HTMLDivElement | null>(null);
+
+    const handleChange = useCallback(
+      (v: string) => {
+        if (searchRef.current && value === undefined) {
+          searchRef.current.value = v;
+          setControlledValue(v);
+        }
+        props?.onChange?.(v);
+      },
+      [props, value]
+    );
+
+    const handleClear = useCallback(() => {
+      onClear?.();
```

Comment:
Har hadde det vært ekstra nice om vi sendte inn hva som trigga "clear": 
```suggestion
      onClear?.({ trigger: "Escape keydown" | "Clear button click" });
```

Reply:
Kunne kanskje vært nice det. Hva tenker du om å bare sende `event` som trigger clear? Kanskje enklere å bruke ved å sende en string for bruker slik du foreslår.

Reply:
Gikk for en begge deler løsning: 
```
type clearEventT =
  | {
      trigger: "Clear button click";
      event: React.MouseEvent<HTMLButtonElement, MouseEvent>;
    }
  | { trigger: "Escape keydown"; event: React.KeyboardEvent<HTMLDivElement> };

onClear?: (e: clearEventT) => void;
```

Reply:
Vurderer om `trigger: "keydown" | "click"` er "enklere" å bruke en eks `Escape keydown` her